source/common/tcp_proxy/factory.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include "common/tcp_proxy/upstream_interface.h"
+
+namespace Envoy {
+
+namespace Upstream {
+class LoadBalancerContext;
+class ClusterManager;
+} // namespace Upstream
+
+namespace TcpProxy {
+
+// TODO(lambdai): move to include path.
```

Comment:
Again as it's a new file, might as well stick where we want it in the long run.
Once it moves to include, it'll need full comments

Reply:
done. Namespace is moved from `TcpProxy` to `Tcp`.